components/Village/Add/index.vue

```diff
@@ -0,0 +1,419 @@
+        this.errors.villageLevel = 'Isian level desa wajib diisi'
+      } else {
+        this.errors.villageLevel = null
+      }
+    },
+    'form.longitude' () {
+      const regexPoint = /^(-?\d+(\.\d+)?)$/
+      if (regexPoint.test(this.form.longitude) || !this.form.latitude.length) {
+        this.errors.longitude = null
+      } else {
+        this.errors.longitude = 'Format isian longitude salah'
+      }
+    },
+    'form.latitude' () {
+      if (this.form.latitude.length === '') {
+        this.form.latitude = ''
+      }
+      const regexPoint = /^(-?\d+(\.\d+)?)$/
+      if (regexPoint.test(this.form.latitude) || !this.form.latitude.length) {
+        this.errors.latitude = null
+      } else {
+        this.errors.latitude = 'Format isian latitude salah'
+      }
+    }
+  },
+  mounted () {
+    this.fetchCity()
+  },
+  methods: {
+    async fetchCity () {
+      try {
+        const response = await this.$axios.get('/cities/suggestion')
+        this.listCity = response.data.data || []
+      } catch {
+        this.listCity = []
+      }
+    },
+    async fetchDistricts (cityId) {
+      try {
+        const response = await this.$axios.get(`/districts/suggestion?city_id=${cityId}`)
+        this.listDistrict = response.data.data || []
+      } catch {
+        this.listDistrict = []
+      }
+    },
+    async addNewVillage (newVillage) {
+      await this.$axios.post('/villages', newVillage, {
+        onUploadProgress: function (progressEvent) {
+          const percentCompleted = parseInt(Math.round((progressEvent.loaded / progressEvent.total) * 100))
+          if (this.showProcessModal) { this.showProcessModal(percentCompleted) }
+        }.bind(this)
+      })
+        .then(() => {
+          this.$store.dispatch('dialog/showDialog', {
+            header: 'Tambah Data Desa Berhasil',
+            title: 'Desa yang Anda tambahkan berhasil disimpan.',
+            message: `DESA DIGITAL - ${this.form.name}`,
+            iconMessage: 'check-mark-circle',
+            iconColor: 'text-green-700',
+            btnLeftVariant: 'primary',
+            btnLeftLabel: 'Saya mengerti',
+            dialogType: 'information',
+            actionBtnLeft: () => this.onModalClose()
+          })
+        })
+        .catch((error) => {
+          const { response: { status, data: { errors } } } = error || {}
+          if (status === 422 && errors) {
+            this.errors.id = errors?.id || null
+            this.errors.name = errors?.name || null
+            this.errors.city_id = errors?.city_id || null
+            this.errors.district_id = errors?.district_id || null
+            this.errors.level = errors?.level || null
+            this.errors.longitude = errors?.longitude || null
+            this.errors.latitude = errors?.latitude || null
+          }
+          this.$store.dispatch('dialog/showDialog', {
+            header: 'Tambah Data Desa Gagal',
+            title: 'Projek yang Anda buat gagal disimpan.',
```

Comment:
minor wording: maybe it's `project` or `proyek`, not projek #cmiiw

Reply:
Thanks for the small details ⭐